File: app/src/main/java/com/stevenfrew/beatprompter/midi/ChannelValue.java
package com.stevenfrew.beatprompter.midi;

public class ChannelValue extends CommandValue {
    ChannelValue(byte channel)
    {
        super(channel);
    }

    @Override
    byte resolve(byte[] arguments, byte channel) {
        return mValue;
    }

    @Override
    public String toString()
    {
        return "#"+mValue;
    }

}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/midi/alias/ChannelValue.kt
package com.stevenfrew.beatprompter.midi.alias

import com.stevenfrew.beatprompter.BeatPrompter
import com.stevenfrew.beatprompter.R

class ChannelValue internal constructor(channel: Byte) : ByteValue(channel) {
	init {
		if (mValue !in 0..15)
			throw ValueException(BeatPrompter.appResources.getString(R.string.invalid_channel_value))
	}

	override fun toString(): String {
		return "#$mValue"
	}
}